docs/provisioner-spec.md

```diff
@@ -0,0 +1,54 @@
+# Provisioner Spec [DRAFT]
+
+## Overview
```

Comment:
Really nice doc! Maybe we want to mention somewhere some examples of provisioners that we could imagine would exist, or that could we plan on implementing? That may be helpful in demystifying the concept more for folks reading the doc.

Reply:
Let's do that in a follow up. I didn't want to cover any more than I had to to meet what we need for #311.